pkg/challenge/challenge.go

```diff
@@ -54,32 +56,48 @@ func (o *OuterChallenge) Marshal() ([]byte, error) {
 	return msgpack.Marshal(o)
 }
 
-func (o *OuterChallenge) Respond(signer crypto.Signer, responseData []byte) ([]byte, error) {
+// Respond creates a response to the challenge. It accepts keys for signing, the second one may be nil.
+func (o *OuterChallenge) Respond(signer crypto.Signer, signer2 crypto.Signer, responseData []byte) ([]byte, error) {
 	if o.innerChallenge == nil {
 		return nil, fmt.Errorf("no inner. unverified?")
 	}
 
-	pubSigningKeyPem, err := echelper.PublicEcdsaKeyToPem(signer.Public().(*ecdsa.PublicKey))
+	pubSigningDer, err := publicEcdsaToDer(signer.Public().(*ecdsa.PublicKey))
 	if err != nil {
-		return nil, fmt.Errorf("marshalling public signing key to pem: %w", err)
+		return nil, fmt.Errorf("marshalling public signing key to der: %w", err)
+	}
+
+	var pubSigningKey2Der []byte
+	if signer2 != nil {
+		pubSigningKey2Der, err = publicEcdsaToDer(signer2.Public().(*ecdsa.PublicKey))
+		if err != nil {
+			return nil, fmt.Errorf("marshalling public signing 2 key to der: %w", err)
+		}
 	}
 
 	innerResponse, err := msgpack.Marshal(InnerResponse{
-		PublicSigningKey: pubSigningKeyPem,
-		ChallengeData:    o.innerChallenge.ChallengeData,
-		ResponseData:     responseData,
-		Timestamp:        time.Now().Unix(),
+		PublicSigningKey:  pubSigningDer,
+		PublicSigningKey2: pubSigningKey2Der,
+		ChallengeData:     o.innerChallenge.ChallengeData,
+		ResponseData:      responseData,
+		Timestamp:         time.Now().Unix(),
 	})
 
 	if err != nil {
 		return nil, fmt.Errorf("marshaling inner box: %w", err)
 	}
 
-	signature, err := echelper.Sign(signer, innerResponse)
+	signature, err := signWithTimeout(signer, innerResponse)
 	if err != nil {
 		return nil, fmt.Errorf("signing challenge: %w", err)
 	}
 
+	var signature2 []byte
+	if signer2 != nil {
+		//nolint: errcheck - we allow nil signer2
+		signature2, _ = signWithTimeout(signer2, innerResponse)
```

Comment:
We don't have a logger here, right? So we can't log the error?

Reply:
correct, we don't thread a logger through to krypto, maybe we should?

Reply:
Eh.